feat: axis labels

zibs opened the pull request:
<!--

Have you read Formidable's Code of Conduct? By filing an Issue, you are expected to comply with it, including treating everyone with respect: https://github.com/FormidableLabs/victory-native-xl/blob/main/CONTRIBUTING.md#contributor-covenant-code-of-conduct

-->

### Description

Adds axis labels

Fixes #350 

<img width="353" alt="Screenshot 2025-01-03 at 1 36 37 PM" src="https://github.com/user-attachments/assets/5aa6b4dd-6071-48d4-977a-4e6ad254e38f" />

<img width="345" alt="Screenshot 2025-01-06 at 8 29 00 AM" src="https://github.com/user-attachments/assets/31055a30-7222-4e6d-b098-d8ca9dd75911" />

#### Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

### How Has This Been Tested?

<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. -->

### Checklist: (Feel free to delete this section upon completion)

- [x] I have included a [changeset](../CONTRIBUTING.md#changesets) if this change will require a version change to one of the packages.
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have run `yarn run check:code` and all checks pass
- [x] I have created a changeset for new features, patches, or major changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

zibs committed:
feat: axis labels

zibs committed:
update example

zibs committed:
docs

zibs committed:
Create slow-cheetahs-fetch.md

zibs committed:
allow multiple y axis labels

zibs committed:
Update YAxis.tsx

keithluchtel commented on example/app/dashed-axes.tsx:
```diff
@@ -42,20 +44,35 @@ export default function DashedAxesPage(props: { segment: string }) {
+            }}
             domain={{ y: [0, 65] }}
-            domainPadding={{ top: 20 }}
+            // domainPadding={{ top: 20 }}
```
do we need this?

keithluchtel commented on lib/src/cartesian/components/XAxis.tsx:
```diff
@@ -105,7 +107,53 @@ export const XAxis = <
+    })();
+
+    // Calculate vertical offset from axis
+    const baseOffset = fontSize * 2 + labelOffset + (yOffset ?? 0);
```
why `fontSize * 2`?

keithluchtel reviewed (commented):
Does adding a label affect the `xScale` and `yScale` range? Do we need to update `transformInputData` at all?

zibs committed:
update for label rotation

for both x and y axis, now we calculate the offset of the axis title by adjusting the axis range to accomodate it.

zibs committed:
use labelWidth for y title offset

zibs committed:
Merge branch 'main' into feat/axis-labels

zibs committed:
Update transformInputData.ts

JulianHinsch commented:
Hello, what's the status of this? We'd love to use labels in our charts :)

zibs commented:
Closing for #672
